src/stacks-hierarchy/Root_LoginPhoneInputScreen.tsx

```diff
@@ -29,7 +31,12 @@ export const Root_LoginPhoneInputScreen = ({
   return (
     <View style={styles.container}>
       <FullScreenHeader
-        leftButton={{type: 'close'}}
+        leftButton={{
+          type:
+            transitionPreset === TransitionPresets.ModalSlideFromBottomIOS
+              ? 'close'
+              : 'back',
+        }}
```

Comment:
I think this is a little stretch, that we change what is rendered to the user based on the transition preset used. I would rather be more precise on it, and have its own argument in the params for the back/close button logic.

in the params for the `Root_PurchaseConfirmationButton` we have the `headerLeftButton` as a prop for this.